automated install/basic-install.sh

```diff
@@ -1990,6 +1990,100 @@ accountForRefactor() {
+            echo "${var} is ${!var}"
+        elif [[ -z "${!var}" ]]; then
+            echo "${var} is not set"
+        else
+            echo "${var} is ${!var}"
+            echo "If ${var} is set in setupVars.conf, valid options are: true or false."
+            exit 1
+        fi
+    done
+    # INSTALL_WEB_INTERFACE must be set to true if INSTALL_WEB_SERVER is set to true
+    if [[ "${INSTALL_WEB_INTERFACE}" == false ]] && [[ "${INSTALL_WEB_SERVER}" == true ]]; then
+        echo "You can install the Web interface without the web server, but you cannot install the web server without the web interface."
+        echo "INSTALL_WEB_SERVER is set to ${INSTALL_WEB_SERVER} and INSTALL_WEB_INTERFACE is set to ${INSTALL_WEB_INTERFACE}."
+        exit 1
+    fi
+    # Regex patterns to match
+    addressMask="^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\/([1-9]|[1-2][0-9]|3[0-2]))$"
+    addressReverse="^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}in-addr\.arpa$"
+    # Validate the IPv4 address and subnet mask
+    if [[ ! ${IPV4_ADDRESS} =~ ${addressMask} ]] ; then
+        echo "${IPV4_ADDRESS} is not a valid address or it is missing the subnet mask"
+        exit 1 
+    else
+        echo "IPv4 ${IPV4_ADDRESS}"
+    fi
+    # Validate the DNS_1 IP addresses
+    if ! valid_ip "${PIHOLE_DNS_1}" ; then
+        echo "${PIHOLE_DNS_1} is not a valid address"
+        exit 1
+    else
+        echo "DNS1 ${PIHOLE_DNS_1}"
+    fi
+    # Validate the DNS_2 IP addresses
+    if ! valid_ip "${PIHOLE_DNS_2}" ; then
+        echo "${PIHOLE_DNS_2} is not a valid address"
+        exit 1
+    else
+        echo "DNS2 ${PIHOLE_DNS_2}"
+    fi
+    # Validate Web UI Layout
+    if [[ "${WEBUIBOXEDLAYOUT}" == "boxed" ]] || [[ "${WEBUIBOXEDLAYOUT}" == "traditional" ]]; then
+            echo "WEBUIBOXEDLAYOUT is ${WEBUIBOXEDLAYOUT}"
+        elif [[ -z "${WEBUIBOXEDLAYOUT}" ]]; then
+            echo "WEBUIBOXEDLAYOUT is not set"
+        else
+            echo "WEBUIBOXEDLAYOUT valid options are: boxed or traditional."
+            exit 1
+        fi
+    # Validate Conditional Forwarding settings
+    if ! valid_ip "${CONDITIONAL_FORWARDING_IP}" ; then
+            echo "CONDITIONAL_FORWARDING_IP is ${CONDITIONAL_FORWARDING_IP}"
+    elif [[ -z "${CONDITIONAL_FORWARDING_IP}" ]]; then
+            echo "CONDITIONAL_FORWARDING_IP is not set"
+    else
+            echo "${CONDITIONAL_FORWARDING_IP} is not a valid IP Address"
+            exit 1
+        fi
+    if [[ "${CONDITIONAL_FORWARDING_REVERSE}" =~ ${addressReverse} ]]; then
+            echo "CONDITIONAL_FORWARDING_REVERSE is ${CONDITIONAL_FORWARDING_REVERSE}"
+    elif [[ -z "${CONDITIONAL_FORWARDING_REVERSE}" ]]; then
+            echo "CONDITIONAL_FORWARDING_REVERSE is not set"
+    else
+            echo "${CONDITIONAL_FORWARDING_REVERSE} is not a valid reverse address"
+            exit 1
+    fi
```

Comment:
CONDITIONAL_FORWARDING has been changed to REV_SERVER. 

`arpa` notation was also replaced with CIDR in that same change.

https://github.com/pi-hole/pi-hole/pull/3882

Reply:
Yeah, nice catch. Although the install script (per the linked pull) will reconfigure `CONDITIONAL_FORWARDING*` vars into `REV_SERVER*` vars - they are [deprecated](https://github.com/pi-hole/docker-pi-hole#deprecated-environment-variables), and we are/should be encouraging folk to use the correct variables, and not let the script do the work of changing the old ones.